lms/djangoapps/program_enrollments/api/linking.py

```diff
@@ -146,6 +172,36 @@ def _get_lms_users(lms_usernames):
     }
 
 
+def unlink_program_enrollment(program_enrollment):
+    """
+    Unlinks CourseEnrollments from the ProgramEnrollment by doing the following for
+    each ProgramCourseEnrollment associated with the Program Enrollment.
+        1. unenrolling the corresponding user from the course
+        2. moving the user into the audit track
```

Comment:
This is laudable learner empathy; I definitely did not think about this & it's a great idea to do.